sdk/identity/identity/test/integration/azureVMUserAssignedTest.spec.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "chai";
+import { Context } from "mocha";
+import { isLiveMode } from "@azure-tools/test-recorder";
+import { ManagedIdentityCredential } from "../../src";
+
+describe("AzureVM UserAssigned Integration test", function () {
+  it("test the Azure VM IMDS endpoint where the MI credential is used.", async function (this: Context) {
+    if (!isLiveMode()) {
+      this.skip();
+    }
+    const userAssignedVM = process.env.IDENTITY_VM_USER_ASSIGNED_MI_CLIENT_ID;
+    if (!userAssignedVM) {
+      console.log("IDENTITY_VM_USER_ASSIGNED_MI_CLIENT_ID is not set");
+      throw new Error("IDENTITY_VM_USER_ASSIGNED_MI_CLIENT_ID is not set");
+    }
+    const credential = new ManagedIdentityCredential({ clientId: userAssignedVM });
+    const accessToken = await credential.getToken("https://management.azure.com//.default");
+    assert.isNotNull(accessToken.token);
+    assert.isDefined(accessToken.token);
```

Comment:
if it's not null it'll definitely be defined. Should this be testing accessToken.expiresOn?

Reply:
I am not sure if for Managed identity it's always set or not. Usually for other credentials it's always there.

Reply:
Yeah actually I had seen this function exists yesterday, but wasn't sure of what it does. Thanks for this.